src/components/SingleArticle.jsx

```diff
@@ -13,18 +13,27 @@ export const SingleArticle = () => {
   const [err, setErr] = useState(null);
   const { article_id } = useParams();
   const [posted, setPosted] = useState(false);
+  const [deleted, setDeleted] = useState(false);
+  const navigate = useNavigate();
   useEffect(() => {
-    api.getArticleById(article_id).then((article) => {
-      setArticle(article);
-      setIsLoading(false);
-    });
+    api
+      .getArticleById(article_id)
+      .then((article) => {
+        setArticle(article);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        setErr("Article not found");
+        setIsLoading(false);
+      });
   }, [article_id]);
+
   useEffect(() => {
     api.getCommentsByArticleId(article_id).then((comments) => {
       setIsLoading(false);
       setComments(comments);
     });
-  }, [article_id, posted]);
+  }, [article_id, posted, deleted]);
```

Comment:
you are fetching all the articles when you delete/post a comment. This is very expensive. Just merge in the the new comment/ filter the one you've just deleted